spec/roundrect.md

```diff
@@ -17,17 +17,39 @@ Proposal
 --------
 
 ```webidl
+
+
 interface mixin CanvasPath {
   // all doubles are unrestricted.
-  void roundRect(double x, double y, double w, double h, sequence<double> radius);
+  void roundRect(unrestricted double x, unrestricted double y, unrestricted double w, unrestricted double h, sequence<(unrestricted double or DOMPoint)> radii);
 };
 ```
 
 `radius` specifies the radius of corners. Each corner is represented by a single radius.
 
-If `radius.length == 1` then all 4 corners have the same length.
+If `radii.length == 1` then all 4 corners have the same length.
+
+If `radii.length == 2` then the first value applies to the top-left and bottom-right corners, and the second value applies to the top-right and bottom-left corners.
+
+If `radii.length == 3` then the first value applies to the top-left corner, the second value applies to the top-right and bottom-left corners, and the third value applies to the bottom-right corner.
+
+If `radii.length == 4` then each corner is specified, in order: top-left, top-right, bottom-right, bottom-left.
+
+If `w` and `h` are both greater than  or equal to 0, or if both are smaller than 0, then the primitive is drawn clockwise. Otherwise, it is drawn conterclockwise.
+
+When `w` is negative, the rounded rectangle is flipped horizontally, which means that the radius values that apply to the left corners, are actual used on the right and vice versa. Similarly, when `h` is negative, the rounded rect is flipped vertically.
+
+When a value `r` in `radii` is a `DOMPoint`, the corresponding corner(s) are drawn as elliptical arcs whose x and y radii are equal to `r.x` and `r.y`, respecively.
+
+When a value `r` in `radii` is a `double`, the corresponding corner(s) are drawn as circular arcs of radius `r`.
+
+When the sum of the radii of two corners of the same edge is greater than the length of the edge, the all the radii of the rounded rectangle are scaled by a factor of len/(r1+r2). If multiple edges have this property, the scale factor of the edge with the smallest scale factor is used.  This is consistent with CSS behaviour.
+
+If a value in `radii` is a negative number, then roudRect() throws an `IndexSizeError` DOM Exception.
```

Comment:
I know it's not in CSS nor even in Skia yet, but what about allowing concave angles? Would that be too much overhead? 
As a web-author that sounds to me like something that could be useful, and I must admit I was a bit surprised / disappointed not later than yesterday when I tried to set a negative radius or DOMPoint going inward and saw it didn't work...